src/org/omegat/filters3/xml/Handler.java

```diff
@@ -637,8 +638,12 @@ private void translateButDontFlash() throws TranslationException {
             // untranslated is written out uncompressed
             if (compressed.equals(translation)) {
                 translation = src;
+                isTranslated = false;
             }
```

Comment:
OmegaT has a concept of "translated" (vs "untranslated") that is independent of whether `compressed.equals(translation)`: you can set "allow translation to be equal to source", in which case the segment can be "translated" even if `compressed.equals(translation) == true`.

But it looks like there is no way to determine the true translation status of the segment at this point, is that right? (Sorry, it's been a long time since I looked at this code.)

If that's the case then I guess this will have to do.

Reply:
I got it.
We should not check  `compressed.equals(translation)` because of useless, and there is no way to detect actual "untranslated" status on OmegaT.

Reply:
> We should not check `compressed.equals(translation)` because of useless

Since the filter is already treating `compressed.equals(translation)` as if it represented the actual status, I think it's OK to use it here as you are doing.

Making the actual status available at this level is an improvement that could be made in the future.

In other words I think this part is OK as it is; no need to change it.